lib/ansible/modules/system/authorized_key.py

```diff
@@ -541,13 +541,13 @@ def enforce_state(module, params):
     error_msg = "Error getting key from: %s"
 
     # if the key is a url, request it and use it as key source
-    if key.startswith("http"):
+    if re.search("^([A-Za-z0-9+.-]+:)?/", key):
```

Comment:
I think I would instead import `urlparse` from `ansible.module_utils.urls` and determine if the `scheme` is populated.

Maybe:

```
if generic_urlparse(urlparse(https_proxy)).scheme:
```

Reply:
This is a good idea, thanks. I've just updated the pull request.
Tested with OpenBSD-CURRENT and python 3.7, BTW.

Reply:
... and now it's Python 2-compatible, ever :)